test/unit/android-helper-specs.js

```diff
@@ -377,12 +377,19 @@ describe('Android Helpers', () => {
     });
   }));
   describe('pushSettingsApp', withMocks({adb}, (mocks) => {
-    it('should install settingsApp', async () => {
-      mocks.adb.expects('install').withExactArgs(settingsApkPath, false).once()
-        .returns('');
+    it('should install settingsApp for real device', async () => {
+      mocks.adb.expects('installOrUpgrade').once()
+        .returns(true);
+      mocks.adb.expects('grantAllPermissions').withExactArgs('io.appium.settings').once()
+        .returns(true);
       await helpers.pushSettingsApp(adb);
       mocks.adb.verify();
     });
+    it('should not install settingsApp for simulator', async () => {
+      mocks.adb.expects('installOrUpgrade').never();
```

Comment:
Why is this expected to be called never? All `helpers.pushSettingsApp` does is call `installOrUpgrade` and `grantAllPermissions` with no conditional. The real device/simulator condition is checked in `initDevice`, so in the test for that is where the distinction should be made. Here you just need the one test, checking that it does indeed call both the intended methods.